doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -1,16 +1,108 @@
 .. _ug_matter_device_security:
 
-Securing production devices
-###########################
+Security
+########
 
 .. contents::
    :local:
-   :depth: 2
+   :depth: 3
+
+Nordic Matter samples leverage security features supported in the |NCS| that can be divided into three major categories:
+
+* Cryptography
+* Trusted storage
+* Securing production devices
+
+In the following sections you will learn more details about each listed category.
+
+Cryptography
+************
+
+Depending on networking backend is use |NCS| Matter samples currently use the following APIs to implement cryptographic operations:
+
+* PSA Cryptography API for Thread networking variant
+* Mbed TLS for Wi-Fi networking variant
+
+Both APIs are integrated in :ref:`nrf_security` library.
+This library offers various configuration possibilities and backends that can be employed to implement :ref:`cryptographic operations <cryptographic_operations_in_ncs>`.
+You can find an overview of cryptography layer configuration supported for each |NCS| Matter-enabled platform in `Matter platforms security support`_ section.
+
+Trusted storage
+***************
+
+The :ref:`Trusted Storage <trusted_storage_in_ncs>` in the |NCS| is a security mechanism designed to securely store and manage sensitive data, such as cryptographic keys, device credentials, and configuration data.
+Such component is essential for IoT devices and it allows to implement secure communication between devices within the Matter ecosystems.
```

Comment:
Doesn't it sound like Matter devices are the only IoT devices possible? :)

Maybe: 
The mechanism is essential for IoT devices, as it allows the implementation secure communication between devices, **for instance**, within the Matter ecosystems.

Reply:
Now that you mention, I'm not sure we need the Matter part for this sentence, since it is stated clearly on Line 41. So just:
```suggestion
The mechanism is essential for IoT devices, as it allows the implementation of secure communication between devices.
```